src/xunit.performance.api/SafeTerminateHandler.cs

```diff
@@ -0,0 +1,78 @@
+﻿using Microsoft.Xunit.Performance.Api.Native.Windows;
+using System;
+
+namespace Microsoft.Xunit.Performance.Api
+{
+    internal sealed class SafeTerminateHandler<T> : IDisposable
+        where T : class, IDisposable
+    {
+        public SafeTerminateHandler(Func<T> createCallback)
+        {
+            BaseDisposableObject = null;
+
+            _HandlerRoutine = (Kernel32.CtrlTypes dwCtrlType) =>
+            {
+                switch (dwCtrlType)
+                {
+                    case Kernel32.CtrlTypes.CTRL_C_EVENT:
+                    case Kernel32.CtrlTypes.CTRL_BREAK_EVENT:
+                    case Kernel32.CtrlTypes.CTRL_CLOSE_EVENT:
+                        BaseDisposableObject?.Dispose();
+                        break;
+                    case Kernel32.CtrlTypes.CTRL_LOGOFF_EVENT:
+                    case Kernel32.CtrlTypes.CTRL_SHUTDOWN_EVENT:
+                        break;
+                }
+                return false;
+            };
+
+            if (!Kernel32.SetConsoleCtrlHandler(_HandlerRoutine, true))
+                throw new Exception("SetConsoleCtrlHandler failed to add handler.");
+
+            BaseDisposableObject = createCallback();
+        }
+
+        public T BaseDisposableObject { get; }
+
+        private readonly Kernel32.PHANDLER_ROUTINE _HandlerRoutine;
```

Comment:
Same as above.
I agree, but I would rather make the split (Windows/Linux) somewhere using a façade to keep the code platform agnostic.